### PR TITLE
Support world joints (joints inside <world> tags)

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
As of https://github.com/gazebosim/sdformat/pull/1117, SDFormat supports joints under world (`//world/joint` in xpath).  This PR adds support for it in gz-sim. 

Note that some systems that operate on joints, such as the `JointPositionController`, require that the system is instantiated inside a model, and therefore, will not work with world joints. It's not clear whether this is a real problem since world joints are most likely to be fixed joints used to attach two models together.

Needs: https://github.com/gazebosim/gz-physics/pull/501

## Test it
A new test `JointsInWorld` has been added to `test/integration/physics_system.cc`.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)~
- [ ] Consider updating Python bindings (if the library has them)~
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/physics/Physics.cc [hunk 1]
```diff
@@ -982,6 +990,7 @@ void PhysicsPrivate::CreateWorldEntities(const EntityComponentManager &_ecm,
         auto worldPtrPhys = this->engine->ConstructWorld(world);
         this->entityWorldMap.AddEntity(_entity, worldPtrPhys);
 
+
```
Comment (ahcorde) [inline src/systems/physics/Physics.cc:993]: remove

File: test/integration/physics_system.cc [hunk 2]
```diff
@@ -2539,3 +2538,86 @@ TEST_F(PhysicsSystemFixture,
...
+//////////////////////////////////////////////////
+/// Tests that joints inside <world> are supported by computing the position of
+/// a pendulum bob from the joint angle. This also tests that commands such as
+/// JointPositionReset work as expected.
+TEST_F(PhysicsSystemFixture, JointsInWorld)
+{
+  ServerConfig serverConfig;
+
+  const auto sdfFile =
+      std::string(PROJECT_SOURCE_PATH) + "/test/worlds/joints_in_world.sdf";
```
Comment (ahcorde) [inline test/integration/physics_system.cc:2551]: `common::joinPath`

File: test/integration/physics_system.cc [hunk 3]
```diff
@@ -2539,3 +2538,86 @@ TEST_F(PhysicsSystemFixture,
...
+              enableComponent<components::JointVelocity>(_ecm, _entity);
+              return true;
+            });
+
+        if (_info.iterations == kIters / 2)
+        {
+          // After half the iterations, reset the joint position and velocity so
+          // that the bob at its equilibrium point and at rest.
+          auto jointEntity = _ecm.EntityByComponents(components::Name("j1"),
+                                                    components::Joint());
```
Comment (ahcorde) [inline test/integration/physics_system.cc:2576]:
```suggestion
                                                     components::Joint());
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/systems/physics/Physics.cc (hunk 1 above)
Comment (azeey) [reply src/systems/physics/Physics.cc]: Done in https://github.com/gazebosim/gz-sim/pull/1949/commits/af117e398bee602f002317d1ae9808a5d1f7add5

File: test/integration/physics_system.cc (hunk 2 above)
Comment (azeey) [reply test/integration/physics_system.cc]: https://github.com/gazebosim/gz-sim/pull/1949/commits/af117e398bee602f002317d1ae9808a5d1f7add5

File: test/integration/physics_system.cc (hunk 3 above)
Comment (azeey) [reply test/integration/physics_system.cc]: https://github.com/gazebosim/gz-sim/pull/1949/commits/af117e398bee602f002317d1ae9808a5d1f7add5

**Comment by azeey** [comment]
@osrf-jenkins run tests please

**Review by marcoag — APPROVED** [review]

**Comment by azeey** [comment]
The `ResetFixture.HandleReset` test has failed at least once in the [past](https://build.osrfoundation.org/view/ign-garden/job/ignition_gazebo-ci-gz-sim7-focal-amd64/110/testReport/(root)/ResetFixture/HandleReset_2/) and doesn't seem  related to this PR. I'll go ahead and merge.

**Pull request merged**